tests/unit/datarobot_drum/drum/test_drum.py

```diff
@@ -0,0 +1,214 @@
+    metadata = {
+        "name": "joe",
+        "type": "training",
+        "targetType": "regression",
+        "environmentID": environment_id,
+        "validation": {"input": "hello"},
+    }
+    yield model_metadata_file_factory(metadata)
+
+
+@pytest.fixture
+def output_dir():
+    with TemporaryDirectory(suffix="output-dir") as dir_name:
+        yield dir_name
+
+
+@pytest.fixture
+def fit_args(temp_metadata, target, output_dir):
+    return [
+        "fit",
+        "--code-dir",
+        temp_metadata,
+        "--input",
+        __file__,
+        "--target",
+        target,
+        "--target-type",
+        "regression",
+        "--output",
+        output_dir,
+    ]
+
+
+@pytest.fixture
+def score_args(this_dir):
+    return [
+        "score",
+        "--code-dir",
+        this_dir,
+        "--input",
+        __file__,
+    ]
+
+
+@pytest.fixture
+def server_args(this_dir):
+    return ["server", "--code-dir", this_dir, "--address", "https://allthedice.com"]
+
+
+@pytest.fixture
+def runtime_factory():
+    with DrumRuntime() as cm_runner_runtime:
+
+        def inner(cli_args):
+            options = get_args_parser_options(cli_args)
+            cm_runner_runtime.options = options
+            cm_runner = CMRunner(cm_runner_runtime)
+            return cm_runner
+
+        yield inner
+
+
+@pytest.fixture
+def mock_input_df(target):
+    with patch.object(CMRunner, "input_df", new_callable=PropertyMock) as mock_prop:
+        data = [[1, 2, 3]] * 100
+        mock_prop.return_value = pd.DataFrame(data, columns=[target, target + "a", target + "b"])
+        yield mock_prop
+
+
+@pytest.fixture
+def mock_get_run_language():
+    with patch.object(
+        CMRunner, "_get_fit_run_language", return_value=RunLanguage.PYTHON
+    ) as mock_func:
+        yield mock_func
+
+
+@pytest.fixture
+def mock_cm_run_test_class(module_under_test):
+    with patch(f"{module_under_test}.CMRunTests") as mock_class:
+        yield mock_class
+
+
+@pytest.mark.usefixtures("mock_input_df", "mock_get_run_language", "mock_cm_run_test_class")
+class TestCMRunnerRunTestPredict:
+    def test_calls_cm_run_test_class_correctly(
+        self, runtime_factory, fit_args, mock_cm_run_test_class, output_dir
+    ):
+        runner = runtime_factory(fit_args)
+        original_options = runner.options
+        original_input = original_options.input
+        target_type = runner.target_type
+        schema_validator = runner.schema_validator
+
+        expected_options = deepcopy(original_options)
+
+        runner.run_test_predict()
+
+        expected_options.input = ANY
+        expected_options.output = os.devnull
+        expected_options.code_dir = output_dir
+
+        mock_cm_run_test_class.assert_called_once_with(
+            expected_options, target_type, schema_validator
+        )
+        actual_options = mock_cm_run_test_class.call_args[0][0]
+        assert actual_options.input != original_input
+
+
+class TestUtilityFunctions:
```

Comment:
this is a new file so that i can start testing CMRunner, but these test should have been in here (they were in utils) and have been moved over